src/pybind/mgr/cli_api/module.py

```diff
@@ -0,0 +1,92 @@
+import json
+import logging
+import threading
+import time
+from functools import partial
+from queue import Queue
+
+from mgr_module import CLICommand, HandleCommandResult, MgrModule
+
+logger = logging.getLogger()
+
+
+class CLI(MgrModule):
+
+    @CLICommand('mgr api get')
+    def api_get(self, arg: str):
+        '''
+        Called by the plugin to fetch named cluster-wide objects from ceph-mgr.
+        :param str data_name: Valid things to fetch are osd_crush_map_text,
+                osd_map, osd_map_tree, osd_map_crush, config, mon_map, fs_map,
+                osd_metadata, pg_summary, io_rate, pg_dump, df, osd_stats,
+                health, mon_status, devices, device <devid>, pg_stats,
+                pool_stats, pg_ready, osd_ping_times.
+        Note:
+            All these structures have their own JSON representations: experiment
+            or look at the C++ ``dump()`` methods to learn about them.
+        '''
+        t1_start = time.time()
+        str_arg = self.get(arg)
+        t1_end = time.time()
+        time_final = (t1_end - t1_start)
+        return HandleCommandResult(0, json.dumps(str_arg), str(time_final))
+
+    @CLICommand('mgr api benchmark get')
+    def api_get_benchmark(self, arg: str, number_of_total_calls: int,
+                          number_of_parallel_calls: int):
+        benchmark_runner = ThreadedBenchmarkRunner(number_of_total_calls, number_of_parallel_calls)
+        benchmark_runner.start(partial(self.get, arg))
+        benchmark_runner.join()
+        stats = benchmark_runner.get_stats()
+        return HandleCommandResult(0, json.dumps(stats), "")
+
+
+class ThreadedBenchmarkRunner:
+    def __init__(self, number_of_total_calls, number_of_parallel_calls):
+        self._number_of_parallel_calls = number_of_parallel_calls
+        self._number_of_total_calls = number_of_total_calls
+        self._threads = []
+        self._jobs: Queue = Queue()
+        self._time = 0.0
+        self._self_time = []
+        self._lock = threading.Lock()
+
+    def start(self, fn):
+        if(self._number_of_total_calls and self._number_of_parallel_calls):
+            for threadID in range(self._number_of_parallel_calls):
```

Comment:
Yup, as @pereman2 suggested, in Python variables are named using the `snake_case` (vs. the `CamelCase` in Typescript for example).
```suggestion
            for thread_id in range(self._number_of_parallel_calls):
```

BTW, [Pylint catches this](http://pylint.pycqa.org/en/latest/user_guide/options.html#predefined-naming-styles). You should add pylint to the .tox where you are running this from. You can copy the pylint settings from the Dashboard .tox.

Reply:
@sebastian-philipp What do you think about this for `src/pybind/mgr/tox.ini ?
This new check would impact other modules and maybe should be done in a different PR (not cache-related).

Reply:
Right, you probably want to add `cli_api` to 

https://github.com/ceph/ceph/blob/91afa17c7309b85b1644bf24e769fd403cf2b22b/src/pybind/mgr/tox.ini#L134-L154

Reply:
@sebastian-philipp Sorry I didn't explain myself. @Waadkh7 already did it, my question was about pylint additional check that @epuertat has mentioned.

Reply:
uh, well. Yeah! Let's add pylint!